src/main/java/io/qmbot/aoc/y2022/Day17.java

```diff
@@ -0,0 +1,249 @@
+package io.qmbot.aoc.y2022;
+
+import io.qmbot.aoc.Puzzle;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Day17 implements Puzzle {
+    @Override
+    public String part1(String input) {
+        String[][] field = new String[24276][9];
+
+        fullField(field);
+
+        int startDirection = 0;
+        int start = field.length - 5;
+        int indexF = 0;
+
+        for (int i = 0; i < 2022; i++) {
+            for (String[] string : field) {
+                if (Arrays.asList(string).contains("7")) {
+                    start = Arrays.asList(field).indexOf(string) - 4;
+                    break;
+                }
+            }
```

Comment:
Lines 20-25 are almost duplicated by lines 33-37. Is it possible to create separate method for them?
Also, if you need to know element's index it is better to iterate by index.